lib/loopdev.c

```diff
@@ -1623,18 +1624,23 @@ int loopcxt_ioctl_blocksize(struct loopdev_cxt *lc, uint64_t blocksize)
 int loopcxt_delete_device(struct loopdev_cxt *lc)
 {
 	int rc, fd = loopcxt_get_fd(lc);
+	int lock_fd = -1;
 
 	if (fd < 0)
 		return -EINVAL;
 
+	lock_fd = open(lc->device, O_RDONLY|O_CLOEXEC);
```

Comment:
This function can be called from libmount, so the fd needs to be closed again.

Reply:
Yes, that's why I think using FD from loopcxt_get_fd() would be better as the context cleanups are all at the end.

Reply:
Will do